lib/Conversion/FIRRTLToHW/LowerToHW.cpp

```diff
@@ -1566,9 +1568,12 @@ LogicalResult FIRRTLLowering::visitDecl(WireOp op) {
   if (resultType.isInteger(0))
     return setLowering(op, Value());
 
-  // Name attr is requires on sv.wire but optional on firrtl.wire.
+  // Name attr is required on sv.wire but optional on firrtl.wire.
   auto nameAttr = op.nameAttr() ? op.nameAttr() : builder.getStringAttr("");
-  return setLoweringTo<sv::WireOp>(op, resultType, nameAttr);
+  // This is not a temporary wire created by the compiler, so attach a symbol
+  // name.
+  // TODO: How to ensure there's no symbol name conflict ?
+  return setLoweringTo<sv::WireOp>(op, resultType, nameAttr, nameAttr);
```

Comment:
I don't think this is correct for the empty firrtl.wire name case.  This will pass a empty string attr that is non-null, which will be treated as a present symbol by the build.  I think you need to pass a `StringAttr()` not a `getStringAttr("")`

Reply:
Thanks, fixed to use `StringAttr()` when name is empty.